### PR TITLE
Mask password input when changing password

### PR DESCRIPTION
**Pull request by pmo3**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

4
---
Mask command line input while changing passwords, so passwords are not displayed in plaintext while typing.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

**Comment by Mcat12** [comment]
How will they set an empty password now?

**Comment by pmo3** [comment]
Entering an empty password and confirming it will still reset the password to be empty. Perhaps we could skip the confirmation step in that case though

**Comment by Mcat12** [comment]
The hash of an empty string coincidentally is not an empty string, so you will need to add some extra logic to handle that case.

**Comment by pmo3** [comment]
Ah you're right. I've changed the logic to set WEBPASSWORD with an unhashed empty string when a user enters a blank password, and to bypass asking for confirmation.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@
 __pycache__
 .cache
 .pullapprove.yml
+Vagrantfile
+.vagrant/
```
Comment (DL6ER) [inline .gitignore]: Please remove these two lines. Your PR should be ready for merging afterwards

Comment (pmo3) [inline .gitignore]: I have made the requested changes

**Comment by pmo3** [comment]
If/when this gets merged I can also open a pull request for the AdminLTE repo to update a few places where the change password option is mentioned in the Dashboard

**Comment by DL6ER** [comment]
@Mcat12 Please re-approve and merge

**Pull request merged**